### DISCUSSION TITLE
[Core Protocol] FailedRelayedMessage  handling

### DISCUSSION THREAD

**Discussion by linkeong88** [ethereum-optimism/developers#381] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I have send a bridge from ethereum main net to base chain using the contract bridgeerc20to, however when it arrived on base it shows FailedRelayedMessage but does not proceed further. Now I have a stuck bridge and I’m unable to go forward and unable to revert. What is the handling for this error? I see previously if the message failed it gets reverted back to main net

https://basescan.org/tx/0x3e2ec42e03d20aa8c3c06a38f1b55a502078dcaa80e5113936387eab47c90f28#eventlog

**Comment by 1195597989** [comment]
You can try to use the relayMessage to attempt to replay the message1.

**Reply by linkeong88** [reply]
It will give the same failure. Is there any other way?

**Comment by hash1go** [comment]
https://github.com/ethereum-optimism/developers/discussions/380#discussioncomment-9311368
According to [Lido's guidance](https://help.lido.fi/en/articles/8545365-bridging-wsteth-to-base), it seems that the deposit's tx of wstETH should be made using [this app](https://superbridge.app/base).
If you issued the transaction through Etherscan, that method is not supported.

If you used the standard bridge, are the transaction details displayed on [this page](https://bridge.base.org/transactions)?
If they are displayed, what is the status?

**Reply by linkeong88** [reply]
The problem is there is no clear instruction on the official bridge that there is another bridge to be used for that token. 
The transaction doesn’t show on the official bridge but it shows on https://superbridge.app/base as successful
![IMG_3855](https://github.com/ethereum-optimism/developers/assets/121043287/007723d9-6f10-45f3-b18b-812941278995)


Irregardless if the transfer is not supported it should be reverted, if not the tokens are still locked by the bridge just like all the other token transfer.

Previous similar issues

https://github.com/ethereum-optimism/optimism/issues/5132

**Reply by hash1go** [reply]
Your transaction did not fail at the deposit stage; it likely failed during the messaging relay beforehand, which may have prevented any recovery processes from occurring.

**Reply by linkeong88** [reply]
I see the same thing happened here but it got recovered

https://github.com/ethereum-optimism/optimism/issues/4794

**Reply by hash1go** [reply]
The past example you provided was successful in relaying the message, hence the revert process was carried out. However, your transaction may have failed in the message relay, which could be why no recovery process has occurred.

**Reply by linkeong88** [reply]
That’s the problem in the first post, when it shows failedrelaymessage what is the next step in handling? Why given the same methods of deposit the message is handled differently and the message was sent using the steps in documentation provided. This question is for the dev team.

So something was send successfully from L1 and locked up correctly on the L1 bridge but unable to receive on L2 with no proper next steps. The lockup happen like the 100s of other token on the L1 base bridge which means it should be able to bridge back to unlock, the token did not vanish into thin air

**Reply by hash1go** [reply]
It seems that the destination for message relays in the deposit process for wstETH is different from the standard in L2. However, your deposit transaction appears to have failed in the message relay because it specified the standard address as the destination.

I will inquire with the OP Team about the recovery process, but it may take a few days.

**Reply by linkeong88** [reply]
Thank you, the previous example I posted also specified the tokens on L2

**Reply by hash1go** [reply]
Hello. I apologize for the delay in providing an update on this matter.
I checked with the engineer at OP Labs, and it seems that your tokens are stuck and cannot be recovered.

**Comment by linkeong88** [comment]
I would hope that the dev actually try to see if there are any solutions rather than just brush it off, as the tokens are now held by the chain owners and anyone bridging to the chain had trusted the credibility of the devs, this will take a big hit on the credibility and drive away users. Furthermore this is not an isolated case or edge case that devs are trying to paint as can be seen from previous postings and other people getting stuck
